src/components/ui/Stock.tsx

```diff
@@ -0,0 +1,27 @@
+import React, { useContext } from "react";
+
+import { useActor } from "@xstate/react";
+import Decimal from "decimal.js-light";
+
+import { Craftable, LimitedItems } from "features/game/types/craftables";
+import { Context } from "features/game/GameProvider";
+
+interface Props {
+    item: Craftable
+}
+
+export const Stock: React.FC<Props> = ({ item }) => {
+  const { gameService } = useContext(Context);
+  const [{ context: { state }}] = useActor(gameService);
+  const stock = state.stock[item.name] || new Decimal(0);
+  const isLimitedItem = item.name in Object.keys(LimitedItems);
```

Comment:
Suggestion to use `hasOwnProperty()` to avoid creating keys array. Might need to typecast `LimitedItems` as `Object`

```
const isLimited = LimitedItems.hasOwnProperty(item.name);
```

Reply:
Yeah I think I missed this. Thanks